### PR TITLE
refactored js and html

### PR DESCRIPTION
**Pull request by Stanimirdim92**
The whole js was refactored as well as fixed some html mistakes.  The factory was refactored as well according to this post: https://samsonasik.wordpress.com/2015/03/31/zend-framework-2-using-__invokepluginmanager-manager-in-services-factory/

Current support is:
Selectors - IE8+ and every other properly created browser
XMLHttpRequest IE 5.5+ ^^ and every other properly created browser
DOMLoading IE7+ and every other properly created browser

I might make few more tweaks these days.

### CODE REVIEW

**Comment by samsonasik** [comment]
thank you. what factory? there is a reason for keep using createService() as commented in blog by lucas, and changed in #26

**Comment by Stanimirdim92** [comment]
Never mind. I returned the old Factory code for the controller.

**Comment by samsonasik** [comment]
ok, review-able? I might check tomorrow

**Comment by Stanimirdim92** [comment]
go ahead.

File: view/zend-developer-tools/toolbar/session-data-list.phtml [hunk 1]
```diff
@@ -71,7 +66,7 @@
                 }
           } else { ?>
         <span class="zdt-toolbar-info">
-                <span class="zdt-detail-label" style="text-transform: none;"><a href="javascript:void(0);">No ZF2 Session Data</a></span>
+                <span class="zdt-detail-label" style="text-transform: none;"><a href="#">No ZF2 Session Data</a></span>
```
Comment (samsonasik) [inline view/zend-developer-tools/toolbar/session-data-list.phtml]: no, there is a reason to use <code>javascript:void(0);</code>, I don't want # appended in url

**Comment by samsonasik** [comment]
there is bug on edit : after edit clicked, and save, next edit don't work ( not show the textfield )

**Comment by samsonasik** [comment]
and bug in other link, after first executed, next click doesn't work

**Comment by samsonasik** [comment]
new bug : delete session per-key doesn't work

File: view/zend-developer-tools/toolbar/session-data-list.phtml (hunk 1 above)
Comment (Stanimirdim92) [inline view/zend-developer-tools/toolbar/session-data-list.phtml]: ok, fixed with e.preventDefault();

File: view/zend-developer-tools/toolbar/session-toolbar.js [hunk 2]
```diff
@@ -14,134 +14,256 @@
...
+            var keysession = trg.parentNode.getAttribute("data-keysession");
+            var params = "containerName="+containerName+"&keysession="+keysession;
+            sanSessionToolbar.postDataWithAjax(baseURL+'/san-session-toolbar/removesession', function (html) {
+                if (html.success) {
+                    // top parent level for this session key - span tag
+                    var elements = doc.querySelector(".san-session-toolbar-info-containerName-"+containerName+"-keysession-"+keysession);
+                    var length = elements.length;
+                    while (length--) {
+                        elements[length].removeChild(elements[length]);
+                    }
```
Comment (samsonasik) [inline view/zend-developer-tools/toolbar/session-toolbar.js]: this is cause the bug in remove session, fixed with : 
```javascript
elements.parentNode.removeChild(elements);
```
I will fix during merge.

**Comment by samsonasik** [comment]
@Stanimirdim92 merged. Thank you ;)

**Comment by Stanimirdim92** [comment]
You are welcome. However, I'm planning another semantics update for the HTML (replacing the <a> tags with buttons and using data-* to target each button instead of classes and/or IDs. This way, when someone changes a class and/or ID in order to fix/apply a new style, he won't break the JS. Also I'm planning some Javascript optimisations and better AJAX handling, but this will happen after I finish my AJAX module. =)

**Comment by samsonasik** [comment]
awesome (y)

Warm Regards,

Abdul Malik Ikhsan


2015-06-24 1:42 GMT+07:00 Stanimir Dimitrov <notifications@github.com>:

**Comment by samsonasik** [comment]
@Stanimirdim92 just note, if you want to do next pull request, start with new branch instead of master, from now on, you can reset and pull latest to sync with current upstream:
```shell
$ git checkout master && git reset --hard HEAD~4
$ git pull https://github.com/samsonasik/SanSessionToolbar.git master
```
and you're safe to do next pull requests ;)

**Comment by Stanimirdim92** [comment]
Yeah, it was a mistake to push directly to master...

**Pull request merged**